mbuild/lib/recipes/polymer.py

```diff
@@ -182,25 +202,30 @@ def build(self, n, sequence="A", add_hydrogens=True):
 
         head_tail = [head_port, tail_port]
 
-        for i, compound in enumerate(self._end_groups):
-            if compound is not None:
-                if self._headtail[i] is not None:
-                    head_tail[i].update_separation(self._headtail[i])
-                self.add(compound)
-                force_overlap(compound, compound.labels["up"], head_tail[i])
-            else:
-                if add_hydrogens:
-                    hydrogen = H()
-                    # Defaut to 1/2 H-C bond len
-                    head_tail[i].update_separation(0.0547)
-                    hydrogen["up"].update_separation(0.0547)
-                    self.add(hydrogen)
-                    force_overlap(hydrogen, hydrogen["up"], head_tail[i])
+        if periodic_axis is not None:
+            self.create_periodic_bond(head_port, tail_port, axis=periodic_axis)
+        else:
+            for i, compound in enumerate(self._end_groups):
+                if compound is not None:
+                    if self._headtail[i] is not None:
+                        head_tail[i].update_separation(self._headtail[i])
+                    self.add(compound)
+                    force_overlap(compound, compound.labels["up"], head_tail[i])
                 else:
-                    # if None, hoist port to polymer level
-                    self.add(
-                        head_tail[i], self._port_labels[i], containment=False
-                    )
+                    if add_hydrogens:
+                        hydrogen = H()
+                        # Defaut to 1/2 H-C bond len
+                        head_tail[i].update_separation(0.0547)
+                        hydrogen["up"].update_separation(0.0547)
+                        self.add(hydrogen)
+                        force_overlap(hydrogen, hydrogen["up"], head_tail[i])
+                    else:
+                        # if None, hoist port to polymer level
+                        self.add(
+                            head_tail[i],
+                            self._port_labels[i],
+                            containment=False,
+                        )
 
         for port in self.all_ports():
             if port not in self.available_ports():
```

Comment:
we need an extra conditional here to make sure we aren't removing the head and tail ports. Maybe something like `and port not in head_tail`